pkg/hubble/relay/pool/client.go

```diff
@@ -47,8 +47,6 @@ func (b GRPCClientConnBuilder) ClientConn(target, hostname string) (poolTypes.Cl
 		return nil, fmt.Errorf("unexpected TLS ServerName %s for %s", hostname, target)
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), b.DialTimeout)
```

Comment:
And again here. We don't use the `DialTimeout` anymore, so let's remove it from the `GRPCClientConnBuilder` too.